File: stack/evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.js
const plus = (a, b) => parseInt(a + b);
const sub = (a, b) => parseInt(a - b);
const multi = (a, b) => parseInt(a * b);
const div = (a, b) => parseInt(a / b);

const PLUS = "+";
const SUB = "-";
const MULTI = "*";
const DIV = "/";

const evalRPN = (tokens) => {
    const stack = [];
    
    for (let i = 0; i < tokens.length; i++) {
        const current_element = tokens[i];
        if (isNaN(+current_element)) {
            if (stack.length > 0) {
                const b = stack.pop();
                const a = stack.pop();        

                if (current_element === PLUS) {
                    const result = plus(a, b);
                    stack.push(result);
                }

                if (current_element === SUB) {
                    const result = sub(a, b);
                    stack.push(result);
                }

                if (current_element === MULTI) {
                    const result = multi(a, b);
                    stack.push(result);
                }

                if (current_element === DIV) {
                    const result = div(a, b);
                    stack.push(result);
                }
            }
        } else {
            stack.push(+current_element);
        }
    }

    const result = stack.at();
    return result;
};

// const tokens = ["2","1","+","3","*"];
// const tokens = ["4","13","5","/","+"];
const tokens = ["10","6","9","3","+","-11","*","/","*","17","+","5","+"]
evalRPN(tokens);

File: stack/evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.java
package org.main;

import org.main.algo.stack.ValidParentheses;

import java.util.Stack;

public class Main {


    public static boolean isNumeric(String str) {
        return str != null && str.matches("-?\\d+(\\.\\d+)?");
    }

    public static int plus(int a, int b){
        return a+b;
    }
    public static int multi(int a, int b){
        return a*b;
    }
    public static int div(int a, int b){
        return a/b;
    }
    public static int minus(int a, int b){
        return a-b;
    }

    final static String PLUS = "+";
    final static String MULTI = "*";
    final static String DIV = "/";
    final static String MINUS = "-";
    public static int evalRPN(String[] tokens){
        Stack stack = new Stack<Integer>();

        for(int i=0 ; i< tokens.length ; i++){
            String current_string = tokens[i];
            if(!isNumeric(current_string)){
                int b = (int) stack.pop();
                int a = (int) stack.pop();

                if(current_string.equals(MINUS)){
                    int result = minus(a,b);
                    stack.push(result);
                }
                if(current_string.equals(PLUS)){
                    int result = plus(a,b);
                    stack.push(result);
                }
                if(current_string.equals(DIV)){
                    int result = div(a,b);
                    stack.push(result);

                }
                if(current_string.equals(MULTI)){
                    int result = multi(a,b);
                    stack.push(result);
                }

            }
            else{
                stack.push(Integer.parseInt(current_string));
            }
        }
     return (int)stack.pop();
    }
    public static void main(String[] args) {
        System.out.println("Hello world!");
        ValidParentheses validator = new ValidParentheses();
        String str = ")(){}";
        boolean isValid = validator.isValid(str);
        System.out.println(isValid);

    }
}